cmd/repo_list.go

```diff
@@ -29,20 +31,36 @@ var repoListCmd = &cobra.Command{
 		if setupErr != nil {
 			return setupErr
 		}
-		_, repoErr := repos.getRepos()
+		list, repoErr := repos.getRepos()
 		if repoErr != nil {
 			return repoErr
 		}
 		repoList, err := repos.listRepos()
 		if err != nil {
 			return err
 		}
-		Info.log("\n", repoList)
+
+		if output == "" {
+			Info.log("\n", repoList)
+		} else if output == "yaml" {
+			result := executeMarshal(&list, yaml.Marshal)
+			Info.log("\n", result)
+		} else if output == "json" {
+			result := executeMarshal(&list, json.Marshal)
+			Info.log("\n", result)
+		}
 		return nil
 	},
 }
 
+func executeMarshal(r interface{}, marshalImpl func(v interface{}) ([]byte, error)) string {
+	bytes, err := marshalImpl(r)
+	if err != nil {
+		Error.log("Could not marshal repository", err)
+	}
+	return string(bytes)
+}
+
 func init() {
 	repoCmd.AddCommand(repoListCmd)
-
```

Comment:
Since we are removing the global flag in `list`, we need to add a local `output` flag to `repo list`.